packages/framework/tests/Feature/DynamicAuthorPagesTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+use Hyde\Facades\Author;
+use Hyde\Testing\TestCase;
+use Illuminate\Support\Facades\Config;
+use Hyde\Facades\Filesystem;
+
+/**
+ * High level test for the dynamic author pages feature.
+ *
+ * @covers \Hyde\Framework\Features\Blogging\BlogPostAuthorPages
+ * @covers \Hyde\Framework\Features\Blogging\DynamicPages\PostAuthorPage
+ * @covers \Hyde\Framework\Features\Blogging\DynamicPages\PostAuthorsPage
+ * @covers \Hyde\Foundation\HydeCoreExtension
+ */
+class DynamicAuthorPagesTest extends TestCase
+{
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->resetSite();
+    }
+
+    public function testAuthorPagesAreGenerated()
+    {
+        $this->setUpTestEnvironment();
+
+        $this->assertSame([
+            '_pages/404.blade.php',
+            '_pages/index.blade.php',
+            '_posts/anonymous_post_1.md',
+            '_posts/guest_post_1.md',
+            '_posts/hyde_post_1.md',
+            '_posts/hyde_post_2.md',
+            '_posts/hyde_post_3.md',
+            '_posts/jane_post_1.md',
+            '_posts/jane_post_2.md',
+            'authors/index',
+            'authors/mr_hyde',
+            'authors/jane_doe',
+            // 'authors/user123', // This user has no posts, so no author page should be created
+            // TODO: 'authors/anonymous', // This will be supported as the author has made a post, and thus exists in the system, even if not in the config
```

Comment:
Need to consider how important this is. It's gonna add a lot of complexity, either we need to find the authors beforehand (which did not go great https://github.com/hydephp/develop/pull/1846 and https://github.com/hydephp/develop/pull/1845), or we need to add them in the listing page (https://github.com/hydephp/develop/pull/1845#issuecomment-2227006201) but this means we also need to create those listing pages. Maybe we'll hold off on this until the future.